src/main/java/org/myrobotlab/service/Polly.java

```diff
@@ -185,6 +190,7 @@ public String[] getKeyNames() {
    */
   @Override
   public AudioData generateAudioData(AudioData audioData, String toSpeak) throws IOException {
+    PollyConfig c = (PollyConfig) config;

```

Comment:
you shouldn't reference the config from here.  instead when you are loading/applying the config, pull the appropriate values into member variables that you can reference .. "autoDetectSsml" in particular, should be a member variable.. and you won't need the config here.  even further, the config object shouldn't exist outside of the load/apply method.  i would prefer to not see it exist at all as a member on the service classes.

Reply:
configuration affects behavior
and denormalization you want to avoid, because it costs maintenance to keep things in sync.
Why not use the same configuration object to affect behavior directly.
This way the mindless copy out from the service, and copy into the service when applying can be avoided